projects/novo-elements/src/elements/query-builder/condition-builder/condition-builder.component.ts

```diff
@@ -187,6 +187,17 @@ export class ConditionBuilderComponent implements OnInit, OnChanges, AfterConten
     this.results$ = Promise.resolve(this.fieldConfig.options);
   }
 
+  reset(values: Record<string, any>): void {
```

Comment:
This seems like a good way of making the value changes, only critiques:

- We might want a different function name. My suggestion for `reset` was under the assumption it was only ever going to clear out the existing values, and return to the default operator with null value. If it's taking values in, then maybe it needs a different name like `setFormValues`.
- If this takes a `values` argument, someone may pass in the `field` key. Have you tested what happens if that is modified as well? (Depending on the component's configuration, the field selector may or may not be hidden)

I am assuming that for the purposes of a cancelable form, you'll be calling this with something like `conditionBuilder.reset({ operator: null, value: null })` when the user clicks Cancel and no previous values have been saved?

Reply:
I think we can reset form values ​​from parent component, no need to do it here, initially I encountered a problem that after I reset the values ​​from the code, the operator template and fields remained the same, because of this I came up with a hack to hide/show the component, so I think now a function that simply resets these templates and initializes them again will be enough, and I will implement the logic of resetting the values ​​in the parent component.